Prettier all the things ✨

barrymcgee opened the pull request.

barrymcgee committed:
Fix up eslint config and add Prettier config

barrymcgee committed:
Prettier all the things ✨

barrymcgee commented on examples/browser/components/statusbar.js:
```diff
@@ -3,20 +3,18 @@
 const PropTypes = require('prop-types');
 const React = require('react');
 
-const {Col, Row} = require('./widgets');
```
These widgets are not being used in this component so I have removed the redundant import.

frankban commented on examples/browser/components/statusbar.js:
```diff
@@ -3,20 +3,18 @@
-  const text = props.logs.slice().reverse().join('\n');
-  return <textarea rows="7" readOnly={true} value={text}></textarea>;
+  const text = props.logs
+    .slice()
```
This change here is a bit weird, given the original line is not that long. Is there a way for disabling this behavior? If not, nevermind.

frankban commented on examples/browser/index.html:
```diff
@@ -13,4 +13,4 @@
 </body>
 
-</html>
+</html>
```
No newline at end of file seems an error.

frankban reviewed (approved):
Looks good with a couple of questions. Also, if we are going to use prettier here, we need to include it in dev dependencies and maybe provide a way for easily running it on the Makefile. Can be a followup branch. Thanks!

barrymcgee committed:
Add Prettier as dev dep

barrymcgee committed:
Add precommit Prettier check

barrymcgee committed:
Correctly name Prettier config file

barrymcgee commented:
@frankban 

- Added Prettier as dev dep
- Added pre-commit step to run Prettier on all staged changes
- Fixed newline omission on `index.html` (My editor was responsible)
- No override option available for chained function calls.

barrymcgee committed:
Ensure prettier.jsxBracketSameLine applies universally
